features/lorawan/system/LoRaWANTimer.cpp

```diff
@@ -62,3 +65,21 @@ void LoRaWANTimeHandler::stop(timer_event_t &obj)
     _queue->cancel(obj.timer_id);
     obj.timer_id = 0;
 }
+
+lorawan_time_t LoRaWANTimeHandler::get_gps_time(void)
+{
+    if (_gps_time == 0) {
+        return _gps_time;
+    }
+
+    return _gps_time + floor((get_current_time() - _monotonic_tick_time) / 1000);
```

Comment:
The `floor` is an unnecessary floating point operation, and anyway, it seems like this should be returning milliseconds, not seconds.

I would suggest a new 64-bit `lorawan_gps_time_t` or something, which is milliseconds since the GPS epoch.